### PR TITLE
Fix bug where targetPort state wasn't unset on change of service

### PR DESCRIPTION
**Pull request by rhamilto**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1641876

### CODE REVIEW

File: frontend/public/components/routes/create-route.tsx [hunk 1]
```diff
@@ -278,7 +280,7 @@ export class CreateRoute extends React.Component<null, CreateRouteState> {
           <div className="form-group co-create-route__target-port">
             <label className="co-required" htmlFor="target-port">Target Port</label>
             {_.isEmpty(portOptions) && <p>Select a service above</p>}
-            {!_.isEmpty(portOptions) && <Dropdown items={portOptions} title="Select target port" dropDownClassName="dropdown--full-width" id="target-port" onChange={this.changeTargetPort} /> }
+            {!_.isEmpty(portOptions) && <Dropdown items={portOptions} title={targetPort ? targetPort : 'Select target port'} dropDownClassName="dropdown--full-width" id="target-port" onChange={this.changeTargetPort} /> }
```
Comment (spadgett) [inline frontend/public/components/routes/create-route.tsx]:
```suggestion
            {!_.isEmpty(portOptions) && <Dropdown items={portOptions} title={targetPort || 'Select target port'} dropDownClassName="dropdown--full-width" id="target-port" onChange={this.changeTargetPort} /> }
```

**Comment by rhamilto** [comment]
Thanks, @spadgett.  Fixed.

**Comment by spadgett** [comment]
jenkins rebuild

**Comment by spadgett** [comment]
We're seeing this a lot :(

```
1) Interacting with the create secret forms Webhook secret : edits webhook secret
   NoSuchElementError: No element found using locator: by.buttonText("Generate")
```

**Comment by spadgett** [comment]
jenkins rebuild

**Comment by spadgett** [comment]
jenkins rebuild

**Comment by spadgett** [comment]
jenkins rebuild

**Comment by spadgett** [comment]
jenkins rebuild

**Comment by spadgett** [comment]
jenkins rebuild

**Pull request merged**